### ISSUE TITLE
Conflict between elementor 3.18.0 and google bunnyfonts replacement plugin

### ISSUE THREAD

**Issue by BackuPs** [elementor/elementor#24512]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When i update to 3.18.0 the footer is suddenly loaded first in the top of the page causing all kinds of jquery errors as jquery is loaded in the header by wordpress.

The moment i revert to 3.17.3 there is not a single issue. All works just fine. This is bizar as the complete website is wrecked by the update.

if i turn off “Optimize Image Loading” by setting it to inactive all is resolved. What does that mean?

https://share.zight.com/Jrumr0AP

It is caused by the plugin replace google fonts with bunny fonts and elementor. They have a conflict

https://wordpress.org/support/plugin/replace-google-fonts-with-bunny-fonts/

The moment this plugin is activated and you are using elementor 3.18.0 the footer of each website will be above the header and the website is totally wrecked.

Tested with a clean wordpress install and hello, astra theme or any elementor theme. They all have the same result.

It is related to a setting in elementor called optimize Image Loading. When set to active (which happens in elementor 3.18.0 by default) the issue starts occuring.

Because the footer is loaded above the header jquery is not loaded and there is a huge amount of jquery errors in the error console.

Checkout these images

https://share.zight.com/qGuY2gqZ

https://share.zight.com/4guRo5Nl

The reason i am using this plugin is because i do not want google tracking on fonts. It is not allowed in the EU. See lawcase in Germany!

But because elementor changed the default setting for “Optimize Image Loading” This issue is now occuring and the replace of the google url does not happen and the website is wrecked. I could of course turn off the setting. But what if i want to use the optimized image loading and the loading of google fonts from bunny.net?

https://share.zight.com/yAunOoRv

Please fix or add google font loading from bunny.net

Thank you

### Steps to reproduce

0) Make sure you have a elementor theme active like hello or astra.
1) Install the https://wordpress.org/support/plugin/replace-google-fonts-with-bunny-fonts/ and activate the plugin
2) Make sure in the elementor settings "optimized image loading" is set to active and save the settings
3) view the front of your website.  The footer will be above the header and the error console will be full of jquery errors..
4) make sure you have elementor 3.18.0 running.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

No needed in this case.

Please add the bunnyfonts url replacement for google fonts default into elementor.  It is not allowed in the EU to load these fonts without permission of the website visitor.

It would be a valuable adjustment to elementor.

Thank you for your time and patience.

**Comment by BackuPs** [comment]
Or add a filter in the frontend.php so we can hook into the google  fonts.googleapis.com  url and replace it with the bunnyfonts url.

Loading fonts from google is not done anymore these days given the GDPR regulations within the EU

Please google on this

https://www.google.com/search?q=germany+lawsuit+loading+google+fonts


https://wptavern.com/german-court-fines-website-owner-for-violating-the-gdpr-by-using-google-hosted-fonts

**Comment by m1ga** [comment]
I would consult a lawyer about Bunny fonts. I would say the same as https://wordpress.org/support/topic/misleading-regarding-gdpr-dsgvo/#post-16894381 that while it is better to have it hosted in the EU you still submit the users IP to a different service and you have to ask for consent first. So just host the fonts locally, don't use an external service. Easy, safe and no trouble

**Comment by BackuPs** [comment]
> I would consult a lawyer about Bunny fonts. I would say the same as https://wordpress.org/support/topic/misleading-regarding-gdpr-dsgvo/#post-16894381 that while it is better to have it hosted in the EU you still submit the users IP to a different service and you have to ask for consent first. So just host the fonts locally, don't use an external service. Easy, safe and no trouble

Btw i am not asking for a legal support comment that has nothing todo with the original issue posted her and which was not answered at all.

**Comment by robwent** [comment]
This image option seems to break Ajax JSON responses and affects multiple plugins.

So far I have seen it break gridbuilder, facetwp, woocommerce product customer list, memberpress drip integration.

**Comment by BackuPs** [comment]
> This image option seems to break Ajax JSON responses and affects multiple plugins.
> 
> So far I have seen it break gridbuilder, facetwp, woocommerce product customer list, memberpress drip integration.

i found also another woo plugin that got issues because of this setting.  The fancy Product designer.  Definite not a very usefull setting.  I wonder why they changed its default from not active to active.  Does not looks like that was a very wise decision.

**Comment by robwent** [comment]
I think this is likely related to the assets, but not the post I remember:
https://github.com/elementor/elementor/issues/10126

It would be 2 lines to add a filter to replace URLs in saved CSS post files.
That way, CDNs could be used for assets in the local files created.

**Comment by BackuPs** [comment]
> I think this is likely related to the assets, but not the post I remember: #10126
> 
> It would be 2 lines to add a filter to replace URLs in saved CSS post files. That way, CDNs could be used for assets in the local files created.

I don't think so...  look at the php code in the module elementor\modules\image-loading-optimization\module.php
